pkg/visibility/api/rest/pending_workloads_lq_test.go

```diff
@@ -0,0 +1,406 @@
+		}
+	)
+
+	scheme := runtime.NewScheme()
+	if err := kueue.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %s", err)
+	}
+	if err := visibility.AddToScheme(scheme); err != nil {
+		t.Fatalf("Failed adding kueue scheme: %s", err)
+	}
+
+	now := time.Now()
+	cases := map[string]struct {
+		clusterQueues []*kueue.ClusterQueue
+		queues        []*kueue.LocalQueue
+		workloads     []*kueue.Workload
+		queryParams   *visibility.PendingWorkloadOptions
+		wantResponse  map[req]*resp
+	}{
+		"single ClusterQueue and single LocalQueue setup with two workloads and default query parameters": {
+			clusterQueues: []*kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue(cqNameA).Obj(),
+			},
+			queues: []*kueue.LocalQueue{
+				utiltesting.MakeLocalQueue(lqNameA, nsNameA).ClusterQueue(cqNameA).Obj(),
+			},
+			workloads: []*kueue.Workload{
+				utiltesting.MakeWorkload("a", nsNameA).Queue(lqNameA).Priority(highPrio).Creation(now).Obj(),
+				utiltesting.MakeWorkload("b", nsNameA).Queue(lqNameA).Priority(lowPrio).Obj(),
+			},
+			queryParams: defaultQueryParams,
+			wantResponse: map[req]*resp{
+				{nsName: nsNameA,
+					queueName:   lqNameA,
+					queryParams: defaultQueryParams,
+				}: {
+					wantPendingWorkloads: []visibility.PendingWorkload{
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Name:      "a",
+								Namespace: nsNameA,
+							},
+							LocalQueueName:         lqNameA,
+							Priority:               highPrio,
+							PositionInClusterQueue: 0,
+							PositionInLocalQueue:   0,
+						},
+						{
+							ObjectMeta: v1.ObjectMeta{
+								Name:      "b",
+								Namespace: nsNameA,
+							},
+							LocalQueueName:         lqNameA,
+							Priority:               lowPrio,
+							PositionInClusterQueue: 1,
+							PositionInLocalQueue:   1,
+						}},
+				},
+			},
+		},
+		"single ClusterQueue and two LocalQueue setup with four workloads and default query parameters": {
+			clusterQueues: []*kueue.ClusterQueue{
+				utiltesting.MakeClusterQueue(cqNameA).Obj(),
+			},
+			queues: []*kueue.LocalQueue{
+				utiltesting.MakeLocalQueue(lqNameA, nsNameA).ClusterQueue(cqNameA).Obj(),
+				utiltesting.MakeLocalQueue(lqNameB, nsNameA).ClusterQueue(cqNameA).Obj(),
+			},
+			workloads: []*kueue.Workload{
+				utiltesting.MakeWorkload("lqA-high-prio", nsNameA).Queue(lqNameA).Priority(highPrio).Creation(now).Obj(),
+				utiltesting.MakeWorkload("lqA-low-prio", nsNameA).Queue(lqNameA).Priority(lowPrio).Creation(now).Obj(),
+				utiltesting.MakeWorkload("lqB-high-prio", nsNameA).Queue(lqNameB).Priority(highPrio).Creation(now.Add(time.Second)).Obj(),
+				utiltesting.MakeWorkload("lqB-low-prio", nsNameA).Queue(lqNameB).Priority(lowPrio).Creation(now.Add(time.Second)).Obj(),
+			},
+			queryParams: defaultQueryParams,
+			wantResponse: map[req]*resp{
+				{
+					nsName:      nsNameA,
+					queueName:   lqNameA,
+					queryParams: defaultQueryParams,
+				}: {wantPendingWorkloads: []visibility.PendingWorkload{
```

Comment:
```suggestion
				}: {
				wantPendingWorkloads: []visibility.PendingWorkload{
```
nit: break the line to make it more readible